### ISSUE TITLE
Feature: onTyping

### ISSUE THREAD

**Issue by smashah** [open-wa/wa-automate-nodejs#578]
Fire user id when user is typing

**Comment by smashah** [comment]
@github-actions run

<details>
<summary>⚡ Release!  ⚡</summary>

```js
(async () => {
function exec(cmd) {
  console.log(execSync(cmd).toString());
}

// Config
const gitUserEmail = "github-actions[bot]@users.noreply.github.com";
const gitUserName = "github-actions[bot]";

exec(`echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > .npmrc`);
exec(`git config --global user.email "${gitUserEmail}"`);
exec(`git config --global user.name "${gitUserName}"`);
exec(`npm i`);
exec(`npm run release-ci`);

//comment on the issue
var result = execSync(`npx auto-changelog -o ./tempchangelog.txt --commit-limit false --template ./compact-keepachangelog.hbs --stdout`).toString();

    await postComment(result);

//create changelog image
exec(`npm run release-image`);
exec(`git commit -a -m 'updated release-image'`);
exec(`git push --force`);
  })();
```
</details>

**Comment by smashah** [comment]
# Changelog

## 🚀 Release [1.9.88](https://github.com/open-wa/wa-automate-nodejs/compare/1.9.87...1.9.88) (2020-06-27)

- Release 1.9.88 [`08c55fc`](https://github.com/open-wa/wa-automate-nodejs/commit/08c55fcd8d15eee9cb4f58def0cc2456527e97c8)
- remove all traces of original middleware #588 [`9b48891`](https://github.com/open-wa/wa-automate-nodejs/commit/9b48891b95a768f89088fb86d05f0c62d82ea918)
- Feature: Implemented `onChatState` #578 [`2b1f0d9`](https://github.com/open-wa/wa-automate-nodejs/commit/2b1f0d9ff2cb4f79b8c40a321f102e49031353ee)

**Comment by smashah** [comment]
```javascript
client.onChatState(x=>concole.log(x))

//fires:
{
  chat: "00000000000-1111111111@g.us",
  user: "22222222222@c.us",
  state:"composing" //other states: available, unavailable, recording
}
```

available is the same as when it shows 'online' in the chat.

This is an insider feature and requires an [Insiders Key](https://gumroad.com/l/BTMt?tier=Insiders%20Program). Learn more here: https://github.com/open-wa/wa-automate-nodejs#license-key
